### DISCUSSION TITLE
What is the code structure that follows #diff in the URL?

### DISCUSSION THREAD

**Discussion by niravbhuva99** [community/community#122741] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

**Example:**

Consider the following URL that highlights changes in a pull request:

`https://github.com/{owner}/{repo}/pull/{pull_request}/files#diff-{sha}`

[https://github.com/facebook/react/pull/28990/files#diff-e6912f3a0bbb760244dd03f6565f02b6c7e7de75656636a13370eb71164f9337](https://github.com/facebook/react/pull/28990/files#diff-e6912f3a0bbb760244dd03f6565f02b6c7e7de75656636a13370eb71164f9337)

I'm uncertain about the code structure that follows `#diff-{sha}` and its functionality.

**Question:**

What is the code structure that follows `#diff` in the URL, and how is it generated?

**Comment by CopperEagle** [comment]
Hey @niravbhuva99 

Github relies on [SHA codes](https://en.wikipedia.org/wiki/Secure_Hash_Algorithms) to uniquely identify an object. Object here really means  anything represented as bytes - like the output from `diff`. (This is a tool which returns the changes between two files, code snippets, etc.) So an object here could be a commit, etc. Github states [here](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/viewing-and-comparing-commits/comparing-commits) they use shortened SHA codes for this. Why SHA? It provides cryptographic strength in that it should be extremely unlikely, that two different objects will end up having the same code ("collision").

You can generate such checksums (codes) for a given file for yourself. If you are on Linux, check out `sha256sum`. You can also use a [website](https://xorbin.com/tools/sha256-hash-calculator) for this.

Hope this answers your question.

**Comment by LangLangBart** [comment, marked as answer]
> and how is it generated?

Related: #55764

**Comment by niravbhuva99** [comment]
@CopperEagle @LangLangBart Thank you for providing this solution, it worked perfectly!
